### PR TITLE
[13.0][IMP] cmis_web_proxy timeout

### PR DESCRIPTION
**Pull request by glitchov**
Add timeout on requests get and post

### CODE REVIEW

**Review by sbidoul — CHANGES_REQUESTED** [review]
Patching requests globally sounds suboptimal as, among possible issues, it will impact other users of the requests library in totally unrelated area of the Odoo process.

So let's pass the timeout argument explicitly, there are not that many.

**Comment by sbidoul** [comment]
Also, I think the timeout configuration should be defined in the [cmis module](https://github.com/OCA/connector-cmis/tree/13.0/cmis) and read here. We probably don't need a different timeout configuration for the cmis proxy and the base cmis module.

**Comment by sbidoul** [comment]
And finally :), can we get the config from an `ir.config_param` ? People who want it in a config file can use [server_environment_ir_config_parameter](https://github.com/OCA/server-env/tree/13.0/server_environment_ir_config_parameter). @glitchov

File: cmis_web_proxy/controllers/cmis.py [hunk 1]
```diff
@@ -130,6 +131,20 @@ def _cmis_proxy_base_url(self):
...
 
+    def _get_requests_timeout(self):
+        """
+        Fetch and return the requests timeout (ir.config_parameter in seconds)
+        Trigger a logged error if something is wrong with the ir.config_parameter
+        :return: timeout (seconds in float)
+        """
+        try:
+            timeout = safe_eval(self.env['ir.config_parameter'].get_param(
+                "cmis.requests_timeout", "FAULTY"))
```
Comment (sbidoul) [inline cmis_web_proxy/controllers/cmis.py:141]: Let's avoid safe_eval when we can. Better use `int()` ... `except ValueError`

Comment (sbidoul) [reply cmis_web_proxy/controllers/cmis.py:141]: or `float()`

Comment (glitchov) [reply cmis_web_proxy/controllers/cmis.py:141]: The reason was that the timeout can also be a tuple of floats (connect timeout and read timeout)

Comment (sbidoul) [reply cmis_web_proxy/controllers/cmis.py:141]: Then the docstring need updating ;) But is that not too fancy ? Is this tuple of timeout something that the requests library supports ? And if we want to use the same config value in the cmis module that uses cmislib, will it support that too ?

Comment (glitchov) [reply cmis_web_proxy/controllers/cmis.py:141]: Maybe it's too fancy but it comes from the requests library when i checked the type of the timeout ;-)
Don't know for cmislib (and did not find any timeout support at all) so we'll keep it simple and i'll follow your wisdom.

File: cmis_web_proxy/controllers/cmis.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from odoo.http import request
 from odoo.exceptions import AccessError
 from odoo.addons.web.controllers import main
+from odoo.tools import safe_eval
```
Comment (sbidoul) [inline cmis_web_proxy/controllers/cmis.py]: unused  import, now

**Review by sbidoul — APPROVED** [review]
Looks good, thanks! (code review only, no test)

File: cmis_web_proxy/controllers/cmis.py (hunk 2 above)
Comment (glitchov) [reply cmis_web_proxy/controllers/cmis.py]: Indeed !

**Review by ThomasBinsfeld — APPROVED** [review]

**Pull request merged**